apps/prairielearn/src/cron/serverLoad.js

```diff
@@ -1,75 +1,66 @@
-const ERR = require('async-stacktrace');
 const async = require('async');
-const AWS = require('aws-sdk');
+const util = require('node:util');
+const { CloudWatch } = require('@aws-sdk/client-cloudwatch');
 
 const { config } = require('../lib/config');
 const sqldb = require('@prairielearn/postgres');
 
-module.exports = {};
+module.exports.run = util.callbackify(async () => {
+  if (!config.runningInEc2) return null;
 
-module.exports.run = function (callback) {
-  if (!config.runningInEc2) return callback(null);
-  const params = [config.groupName, config.serverLoadAverageIntervalSec];
-  sqldb.call('server_loads_current', params, (err, result) => {
-    if (ERR(err, callback)) return;
-    if (result.rowCount === 0) return callback(null); // nothing to report
-    const cloudwatch = new AWS.CloudWatch(config.awsServiceGlobalOptions);
-    async.each(
-      result.rows,
-      (row, callback) => {
-        const params = {
-          Namespace: 'PrairieLearn',
-          MetricData: [],
-        };
-        const dimensions = [
-          { Name: 'Server Group', Value: config.groupName },
-          { Name: 'Job Type', Value: row.job_type },
-        ];
-        params.MetricData.push(
-          ...[
-            {
-              MetricName: 'InstanceCount',
-              Dimensions: dimensions,
-              StorageResolution: 1,
-              Timestamp: row.timestamp_formatted,
-              Unit: 'Count',
-              Value: row.instance_count,
-            },
-            {
-              MetricName: 'CurrentJobs',
-              Dimensions: dimensions,
-              StorageResolution: 1,
-              Timestamp: row.timestamp_formatted,
-              Unit: 'Count',
-              Value: row.current_jobs,
-            },
-            {
-              MetricName: 'MaxJobs',
-              Dimensions: dimensions,
-              StorageResolution: 1,
-              Timestamp: row.timestamp_formatted,
-              Unit: 'Count',
-              Value: row.max_jobs,
-            },
-            {
-              MetricName: 'LoadPercentage',
-              Dimensions: dimensions,
-              StorageResolution: 1,
-              Timestamp: row.timestamp_formatted,
-              Unit: 'Percent',
-              Value: row.load_perc,
-            },
-          ]
-        );
-        cloudwatch.putMetricData(params, function (err, _data) {
-          if (ERR(err, callback)) return;
-          callback(null);
-        });
-      },
-      (err) => {
-        if (ERR(err, callback)) return;
-        callback(null);
-      }
-    );
+  const result = await sqldb.callAsync('server_loads_current', [
+    config.groupName,
+    config.serverLoadAverageIntervalSec,
+  ]);
+  if (result.rowCount === 0) {
+    // Nothing to report.
+    return;
+  }
+  const cloudwatch = new CloudWatch({
+    region: config.awsRegion,
+    ...config.awsServiceGlobalOptions,
```

Comment:
We could have a little `makeAwsSdkOptions()` helper to do this little song and dance? I also reached out to @trombonekenny to see if we could get rid of `config.awsServiceGlobalOptions` entirely.

Reply:
Sounds like we can indeed lose `awsServiceGlobalOptions`.

Reply:
After talking with Dave, I'm going to opt to keep this for the time being. I'll go the `make...` route.